### DISCUSSION TITLE
Serial communication issues

### DISCUSSION THREAD

**Discussion by billyd60** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1873] [Q&A] [answered]
Ok I've been struggling with this issue for some time. I have an Archim2 mainboard on Marlin 2.0.x bugfix (4/8/21) and the TFT70 V3 with .27 firmware.

The problem: Everything functions perfectly except when printing via touchscreen mode. In this mode the data is sent to the mainboard by the serial connection. The menus and all functions in the menus seem to function well. Upon first startup it sometimes doesn't recognize one command "M305" (I have a PT100 temp sensor and amplifier). But it reads the temp perfectly. When I attempt a print via SD card the print seems to work normally for a period of time but then the printer starts making random moves on occasion and then continues to print normally. Every once in awhile it will freeze when printing in this mode.

It's important to remember that printing via SD in Marlin Mode is flawless. This is the flat ribbon cable connection.

So we know a few things based on these facts.
1. The printer can print things perfectly fine.
2. The TFT70 in Marlin Mode functions perfectly.
3. Serial communication is established but not always stable.

I have tried the following to eliminate the problem without success
1. Adjusted baud rate from 250000 to 115200
2. Made a new serial cable 2 times, both heavily shielded with the shield to attached to ground at one end as is proper.
3. Changed the routing of the serial cable to avoid any possible interference.
4. Removed the ribbon cables and ran via serial only
5. Removes the 5v power from the serial cable (and no ribbon cables) and supplied 5v via a battery powered regulator to the TFT power input connector.
6. Made recommended changes to the serial port connection in Marlin based on comments from the community.

I am out of ideas any help is greatly appreciated.

Attached are the config files for my Marlin2 firmware.
[configH_and_configadvH_files_revB.zip](https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware/files/6367784/configH_and_configadvH_files_revB.zip)

**Comment by billyd60** [comment, marked as answer]
I resolved this issue with the latest version of Marlin and I missed some setup steps per the readme here on BTT's github.

**Reply by jafarniknam** [reply]
HI . 
same problem . I use MKS GEN V1.4 + MKS TFT32 with bigtreetech touch screen firmware but no printer connection : "No printer attached " message 
Of course everything is OK with original MKS firmware  (REPRAP FW) and commands send to motherboard without any problem . 
can you help me ? 
thanks and Sorry for my English !!

**Reply by tincore** [reply]
Hi, 

I have the same issue. 

Using BTT TFT35-E3 v3.0 with  BTT SKR Mini E3 v3.0

Running on up to date Marlin and also updated to latest release (Feb/2023) TFT firmware.

I also tried different baud rates, cables, routing...

Printing from board SD has no problem, printing from TFT SD or USB freezes **depending on print speed**. 

In my case freezes happens when  speeds are greater than about 90/100 mm/sec. Speed is definitively a factor.

Sometimes before the freeze I get some random move or message about bad command on screen. it seems like transmitted commands are corrupted.

**Reply by tincore** [reply]
I'm on Marlin bugfix-2.1.x. @billyd60 I see that you mentioned bugfix-2.0.x  in 2021. Are you currently in a recent version of that branch? I see a 2.0.x commit from Oct 2022.

**Reply by billyd60** [reply]
Go to here: https://github.com/bigtreetech/BIGTREETECH-TouchScreenFirmware and scroll down the page and keep reading everything. There are many settings that you must put in place to get the serial side (and the entire tft) working properly. Do everything per those instructions. I am currently on December 26th of 2021 version 2.0.x bugfix. I've been thinking about updating everything but it's working and I don't need any other features than I already have. The printer is pretty old at this point doesn't get used as much as my other printers.
